// @aws-cdk/aws-amplify-alpha/lib/app.js
"use strict";
var __runInitializers = (this && this.__runInitializers) || function (thisArg, initializers, value) {
    var useValue = arguments.length > 2;
    for (var i = 0; i < initializers.length; i++) {
        value = useValue ? initializers[i].call(thisArg, value) : initializers[i].call(thisArg);
    }
    return useValue ? value : void 0;
};
var __esDecorate = (this && this.__esDecorate) || function (ctor, descriptorIn, decorators, contextIn, initializers, extraInitializers) {
    function accept(f) { if (f !== void 0 && typeof f !== "function") throw new TypeError("Function expected"); return f; }
    var kind = contextIn.kind, key = kind === "getter" ? "get" : kind === "setter" ? "set" : "value";
    var target = !descriptorIn && ctor ? contextIn["static"] ? ctor : ctor.prototype : null;
    var descriptor = descriptorIn || (target ? Object.getOwnPropertyDescriptor(target, contextIn.name) : {});
    var _, done = false;
    for (var i = decorators.length - 1; i >= 0; i--) {
        var context = {};
        for (var p in contextIn) context[p] = p === "access" ? {} : contextIn[p];
        for (var p in contextIn.access) context.access[p] = contextIn.access[p];
        context.addInitializer = function (f) { if (done) throw new TypeError("Cannot add initializers after decoration has completed"); extraInitializers.push(accept(f || null)); };
        var result = (0, decorators[i])(kind === "accessor" ? { get: descriptor.get, set: descriptor.set } : descriptor[key], context);
        if (kind === "accessor") {
            if (result === void 0) continue;
            if (result === null || typeof result !== "object") throw new TypeError("Object expected");
            if (_ = accept(result.get)) descriptor.get = _;
            if (_ = accept(result.set)) descriptor.set = _;
            if (_ = accept(result.init)) initializers.unshift(_);
        }
        else if (_ = accept(result)) {
            if (kind === "field") initializers.unshift(_);
            else descriptor[key] = _;
        }
    }
    if (target) Object.defineProperty(target, contextIn.name, descriptor);
    done = true;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuildComputeType = exports.CacheConfigType = exports.Platform = exports.CustomRule = exports.RedirectStatus = exports.App = void 0;
const jsiiDeprecationWarnings = require("../.warnings.jsii.js");
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const aws_amplify_1 = require("aws-cdk-lib/aws-amplify");
const iam = require("aws-cdk-lib/aws-iam");
const core_1 = require("aws-cdk-lib/core");
const helpers_internal_1 = require("aws-cdk-lib/core/lib/helpers-internal");
const metadata_resource_1 = require("aws-cdk-lib/core/lib/metadata-resource");
const prop_injectable_1 = require("aws-cdk-lib/core/lib/prop-injectable");
const branch_1 = require("./branch");
const domain_1 = require("./domain");
const utils_1 = require("./utils");
/**
 * An Amplify Console application
 */
let App = (() => {
    let _classDecorators = [prop_injectable_1.propertyInjectable];
    let _classDescriptor;
    let _classExtraInitializers = [];
    let _classThis;
    let _classSuper = core_1.Resource;
    let _instanceExtraInitializers = [];
    let _addCustomRule_decorators;
    let _addEnvironment_decorators;
    let _addAutoBranchEnvironment_decorators;
    let _addBranch_decorators;
    let _addDomain_decorators;
    var App = class extends _classSuper {
        static { _classThis = this; }
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(_classSuper[Symbol.metadata] ?? null) : void 0;
            _addCustomRule_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            _addEnvironment_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            _addAutoBranchEnvironment_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            _addBranch_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            _addDomain_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            __esDecorate(this, null, _addCustomRule_decorators, { kind: "method", name: "addCustomRule", static: false, private: false, access: { has: obj => "addCustomRule" in obj, get: obj => obj.addCustomRule }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _addEnvironment_decorators, { kind: "method", name: "addEnvironment", static: false, private: false, access: { has: obj => "addEnvironment" in obj, get: obj => obj.addEnvironment }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _addAutoBranchEnvironment_decorators, { kind: "method", name: "addAutoBranchEnvironment", static: false, private: false, access: { has: obj => "addAutoBranchEnvironment" in obj, get: obj => obj.addAutoBranchEnvironment }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _addBranch_decorators, { kind: "method", name: "addBranch", static: false, private: false, access: { has: obj => "addBranch" in obj, get: obj => obj.addBranch }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _addDomain_decorators, { kind: "method", name: "addDomain", static: false, private: false, access: { has: obj => "addDomain" in obj, get: obj => obj.addDomain }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(null, _classDescriptor = { value: _classThis }, _classDecorators, { kind: "class", name: _classThis.name, metadata: _metadata }, null, _classExtraInitializers);
            App = _classThis = _classDescriptor.value;
            if (_metadata) Object.defineProperty(_classThis, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        static [JSII_RTTI_SYMBOL_1] = { fqn: "@aws-cdk/aws-amplify-alpha.App", version: "2.248.0-alpha.0" };
        /** Uniquely identifies this class. */
        static PROPERTY_INJECTION_ID = '@aws-cdk.aws-amplify-alpha.App';
        /**
         * Import an existing application
         */
        static fromAppId(scope, id, appId) {
            class Import extends core_1.Resource {
                appId = appId;
            }
            return new Import(scope, id);
        }
        appId = __runInitializers(this, _instanceExtraInitializers);
        /**
         * The name of the application
         *
         * @attribute
         */
        appName;
        /**
         * The ARN of the application
         *
         * @attribute
         */
        arn;
        /**
         * The default domain of the application
         *
         * @attribute
         */
        defaultDomain;
        /**
         * The principal to grant permissions to
         */
        grantPrincipal;
        /**
         * The IAM role for an SSR app.
         */
        computeRole;
        /**
         * The platform of the app
         */
        platform;
        customRules;
        environmentVariables;
        autoBranchEnvironmentVariables;
        constructor(scope, id, props) {
            super(scope, id);
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_AppProps(props);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, App);
                }
                throw error;
            }
            // Enhanced CDK Analytics Telemetry
            (0, metadata_resource_1.addConstructMetadata)(this, props);
            this.customRules = props.customRules || [];
            this.environmentVariables = props.environmentVariables || {};
            this.autoBranchEnvironmentVariables = props.autoBranchCreation && props.autoBranchCreation.environmentVariables || {};
            const role = props.role || new iam.Role(this, 'Role', {
                assumedBy: new iam.ServicePrincipal('amplify.amazonaws.com'),
            });
            this.grantPrincipal = role;
            let computedRole;
            const appPlatform = props.platform || Platform.WEB;
            const isSSR = (0, utils_1.isServerSideRendered)(appPlatform);
            if (props.computeRole) {
                if (!isSSR) {
                    throw new core_1.ValidationError((0, helpers_internal_1.lit) `InvalidComputeRolePlatform`, '`computeRole` can only be specified for `Platform.WEB_COMPUTE` or `Platform.WEB_DYNAMIC`.', this);
                }
                computedRole = props.computeRole;
            }
            else if (isSSR) {
                computedRole = new iam.Role(this, 'ComputeRole', {
                    assumedBy: new iam.ServicePrincipal('amplify.amazonaws.com'),
                });
            }
            this.computeRole = computedRole;
            const sourceCodeProviderOptions = props.sourceCodeProvider?.bind(this);
            this.platform = appPlatform;
            const app = new aws_amplify_1.CfnApp(this, 'Resource', {
                accessToken: sourceCodeProviderOptions?.accessToken?.unsafeUnwrap(), // Safe usage
                autoBranchCreationConfig: props.autoBranchCreation && {
                    autoBranchCreationPatterns: props.autoBranchCreation.patterns,
                    basicAuthConfig: props.autoBranchCreation.basicAuth
                        ? props.autoBranchCreation.basicAuth.bind(this, 'BranchBasicAuth')
                        : { enableBasicAuth: false },
                    buildSpec: props.autoBranchCreation.buildSpec && props.autoBranchCreation.buildSpec.toBuildSpec(),
                    enableAutoBranchCreation: true,
                    enableAutoBuild: props.autoBranchCreation.autoBuild ?? true,
                    environmentVariables: core_1.Lazy.any({ produce: () => (0, utils_1.renderEnvironmentVariables)(this.autoBranchEnvironmentVariables) }, { omitEmptyArray: true }),
                    enablePullRequestPreview: props.autoBranchCreation.pullRequestPreview ?? true,
                    pullRequestEnvironmentName: props.autoBranchCreation.pullRequestEnvironmentName,
                    stage: props.autoBranchCreation.stage,
                },
                enableBranchAutoDeletion: props.autoBranchDeletion,
                basicAuthConfig: props.basicAuth
                    ? props.basicAuth.bind(this, 'AppBasicAuth')
                    : { enableBasicAuth: false },
                buildSpec: props.buildSpec && props.buildSpec.toBuildSpec(),
                cacheConfig: props.cacheConfigType ? { type: props.cacheConfigType } : undefined,
                computeRoleArn: this.computeRole?.roleArn,
                customRules: core_1.Lazy.any({ produce: () => this.customRules }, { omitEmptyArray: true }),
                description: props.description,
                environmentVariables: core_1.Lazy.any({ produce: () => (0, utils_1.renderEnvironmentVariables)(this.environmentVariables) }, { omitEmptyArray: true }),
                iamServiceRole: role.roleArn,
                name: props.appName || this.node.id,
                oauthToken: sourceCodeProviderOptions?.oauthToken?.unsafeUnwrap(), // Safe usage
                repository: sourceCodeProviderOptions?.repository,
                customHeaders: props.customResponseHeaders && props.customResponseHeaders.length > 0
                    ? renderCustomResponseHeaders(props.customResponseHeaders, this)
                    : undefined,
                platform: appPlatform,
                jobConfig: props.buildComputeType ? { buildComputeType: props.buildComputeType } : undefined,
            });
            this.appId = app.attrAppId;
            this.appName = app.attrAppName;
            this.arn = app.attrArn;
            this.defaultDomain = app.attrDefaultDomain;
        }
        /**
         * Adds a custom rewrite/redirect rule to this application
         */
        addCustomRule(rule) {
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_CustomRule(rule);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, this.addCustomRule);
                }
                throw error;
            }
            this.customRules.push(rule);
            return this;
        }
        /**
         * Adds an environment variable to this application.
         *
         * All environment variables that you add are encrypted to prevent rogue
         * access so you can use them to store secret information.
         */
        addEnvironment(name, value) {
            this.environmentVariables[name] = value;
            return this;
        }
        /**
         * Adds an environment variable to the auto created branch.
         *
         * All environment variables that you add are encrypted to prevent rogue
         * access so you can use them to store secret information.
         */
        addAutoBranchEnvironment(name, value) {
            this.autoBranchEnvironmentVariables[name] = value;
            return this;
        }
        /**
         * Adds a branch to this application
         */
        addBranch(id, options = {}) {
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_BranchOptions(options);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, this.addBranch);
                }
                throw error;
            }
            return new branch_1.Branch(this, id, {
                ...options,
                app: this,
            });
        }
        /**
         * Adds a domain to this application
         */
        addDomain(id, options = {}) {
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_DomainOptions(options);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, this.addDomain);
                }
                throw error;
            }
            return new domain_1.Domain(this, id, {
                ...options,
                app: this,
                autoSubDomainIamRole: this.grantPrincipal,
            });
        }
        static {
            __runInitializers(_classThis, _classExtraInitializers);
        }
    };
    return App = _classThis;
})();
exports.App = App;
/**
 * The status code for a URL rewrite or redirect rule.
 */
var RedirectStatus;
(function (RedirectStatus) {
    /**
     * Rewrite (200)
     */
    RedirectStatus["REWRITE"] = "200";
    /**
     * Permanent redirect (301)
     */
    RedirectStatus["PERMANENT_REDIRECT"] = "301";
    /**
     * Temporary redirect (302)
     */
    RedirectStatus["TEMPORARY_REDIRECT"] = "302";
    /**
     * Not found (404)
     */
    RedirectStatus["NOT_FOUND"] = "404";
    /**
     * Not found rewrite (404)
     */
    RedirectStatus["NOT_FOUND_REWRITE"] = "404-200";
})(RedirectStatus || (exports.RedirectStatus = RedirectStatus = {}));
/**
 * Custom rewrite/redirect rule for an Amplify App.
 *
 * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
 */
class CustomRule {
    static [JSII_RTTI_SYMBOL_1] = { fqn: "@aws-cdk/aws-amplify-alpha.CustomRule", version: "2.248.0-alpha.0" };
    /**
     * Sets up a 200 rewrite for all paths to `index.html` except for path
     * containing a file extension.
     */
    static SINGLE_PAGE_APPLICATION_REDIRECT = new CustomRule({
        source: '</^[^.]+$/>',
        target: '/index.html',
        status: RedirectStatus.REWRITE,
    });
    /**
     * The source pattern for a URL rewrite or redirect rule.
     *
     * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
     */
    source;
    /**
     * The target pattern for a URL rewrite or redirect rule.
     *
     * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
     */
    target;
    /**
     * The status code for a URL rewrite or redirect rule.
     *
     * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
     *
     * @default PERMANENT_REDIRECT
     */
    status;
    /**
     * The condition for a URL rewrite or redirect rule, e.g. country code.
     *
     * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
     *
     * @default - no condition
     */
    condition;
    constructor(options) {
        try {
            jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_CustomRuleOptions(options);
        }
        catch (error) {
            if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                Error.captureStackTrace(error, CustomRule);
            }
            throw error;
        }
        this.source = options.source;
        this.target = options.target;
        this.status = options.status;
        this.condition = options.condition;
    }
}
exports.CustomRule = CustomRule;
/**
 * Renders custom response headers to YAML format.
 *
 * @param customHeaders - Array of custom headers. Must not be empty.
 * @param scope - Construct scope for error reporting
 * @returns YAML string representation of custom headers
 *
 * @internal
 */
function renderCustomResponseHeaders(customHeaders, scope) {
    // Defensive assertion - should never happen due to call site validation
    if (customHeaders.length === 0) {
        throw new core_1.ValidationError((0, helpers_internal_1.lit) `EmptyCustomResponseHeaders`, 'renderCustomResponseHeaders called with empty array', scope);
    }
    const hasAppRoot = customHeaders[0].appRoot !== undefined;
    const yaml = [hasAppRoot ? 'applications:' : 'customHeaders:'];
    for (const customHeader of customHeaders) {
        if ((customHeader.appRoot !== undefined) !== hasAppRoot) {
            throw new core_1.ValidationError((0, helpers_internal_1.lit) `InconsistentAppRoot`, 'appRoot must be either be present or absent across all custom response headers', scope);
        }
        const baseIndentation = ' '.repeat(hasAppRoot ? 6 : 2);
        if (hasAppRoot) {
            yaml.push(`  - appRoot: ${customHeader.appRoot}`);
            yaml.push('    customHeaders:');
        }
        yaml.push(`${baseIndentation}- pattern: "${customHeader.pattern}"`);
        yaml.push(`${baseIndentation}  headers:`);
        for (const [key, value] of Object.entries(customHeader.headers)) {
            yaml.push(`${baseIndentation}    - key: "${key}"`);
            yaml.push(`${baseIndentation}      value: "${value}"`);
        }
    }
    return `${yaml.join('\n')}\n`;
}
/**
 * Available hosting platforms to use on the App.
 */
var Platform;
(function (Platform) {
    /**
     * WEB - Used to indicate that the app is hosted using only static assets.
     */
    Platform["WEB"] = "WEB";
    /**
     * WEB_COMPUTE - Used to indicate the app is hosted using a combination of
     * server side rendered and static assets.
     */
    Platform["WEB_COMPUTE"] = "WEB_COMPUTE";
    /**
     * WEB_DYNAMIC - Used to indicate the app is hosted using a fully dynamic architecture, where requests are processed at runtime by backend compute services.
     */
    Platform["WEB_DYNAMIC"] = "WEB_DYNAMIC";
})(Platform || (exports.Platform = Platform = {}));
/**
 * The type of cache configuration to use for an Amplify app.
 */
var CacheConfigType;
(function (CacheConfigType) {
    /**
     * AMPLIFY_MANAGED - Automatically applies an optimized cache configuration
     * for your app based on its platform, routing rules, and rewrite rules.
     */
    CacheConfigType["AMPLIFY_MANAGED"] = "AMPLIFY_MANAGED";
    /**
     * AMPLIFY_MANAGED_NO_COOKIES - The same as AMPLIFY_MANAGED,
     * except that it excludes all cookies from the cache key.
     */
    CacheConfigType["AMPLIFY_MANAGED_NO_COOKIES"] = "AMPLIFY_MANAGED_NO_COOKIES";
})(CacheConfigType || (exports.CacheConfigType = CacheConfigType = {}));
/**
 * Specifies the size of the build instance.
 *
 * @link https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-instance.html
 */
var BuildComputeType;
(function (BuildComputeType) {
    /**
     * vCPUs: 4, Memory: 8 GiB, Disk space: 128 GB
     */
    BuildComputeType["STANDARD_8GB"] = "STANDARD_8GB";
    /**
     * vCPUs: 8, Memory: 16 GiB, Disk space: 128 GB
     */
    BuildComputeType["LARGE_16GB"] = "LARGE_16GB";
    /**
     * vCPUs: 36, Memory: 72 GiB, Disk space: 256 GB
     */
    BuildComputeType["XLARGE_72GB"] = "XLARGE_72GB";
})(BuildComputeType || (exports.BuildComputeType = BuildComputeType = {}));
