// @aws-cdk/aws-amplify-alpha/lib/branch.js
"use strict";
var __runInitializers = (this && this.__runInitializers) || function (thisArg, initializers, value) {
    var useValue = arguments.length > 2;
    for (var i = 0; i < initializers.length; i++) {
        value = useValue ? initializers[i].call(thisArg, value) : initializers[i].call(thisArg);
    }
    return useValue ? value : void 0;
};
var __esDecorate = (this && this.__esDecorate) || function (ctor, descriptorIn, decorators, contextIn, initializers, extraInitializers) {
    function accept(f) { if (f !== void 0 && typeof f !== "function") throw new TypeError("Function expected"); return f; }
    var kind = contextIn.kind, key = kind === "getter" ? "get" : kind === "setter" ? "set" : "value";
    var target = !descriptorIn && ctor ? contextIn["static"] ? ctor : ctor.prototype : null;
    var descriptor = descriptorIn || (target ? Object.getOwnPropertyDescriptor(target, contextIn.name) : {});
    var _, done = false;
    for (var i = decorators.length - 1; i >= 0; i--) {
        var context = {};
        for (var p in contextIn) context[p] = p === "access" ? {} : contextIn[p];
        for (var p in contextIn.access) context.access[p] = contextIn.access[p];
        context.addInitializer = function (f) { if (done) throw new TypeError("Cannot add initializers after decoration has completed"); extraInitializers.push(accept(f || null)); };
        var result = (0, decorators[i])(kind === "accessor" ? { get: descriptor.get, set: descriptor.set } : descriptor[key], context);
        if (kind === "accessor") {
            if (result === void 0) continue;
            if (result === null || typeof result !== "object") throw new TypeError("Object expected");
            if (_ = accept(result.get)) descriptor.get = _;
            if (_ = accept(result.set)) descriptor.set = _;
            if (_ = accept(result.init)) initializers.unshift(_);
        }
        else if (_ = accept(result)) {
            if (kind === "field") initializers.unshift(_);
            else descriptor[key] = _;
        }
    }
    if (target) Object.defineProperty(target, contextIn.name, descriptor);
    done = true;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Branch = void 0;
const jsiiDeprecationWarnings = require("../.warnings.jsii.js");
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const aws_amplify_1 = require("aws-cdk-lib/aws-amplify");
const iam = require("aws-cdk-lib/aws-iam");
const core_1 = require("aws-cdk-lib/core");
const helpers_internal_1 = require("aws-cdk-lib/core/lib/helpers-internal");
const metadata_resource_1 = require("aws-cdk-lib/core/lib/metadata-resource");
const prop_injectable_1 = require("aws-cdk-lib/core/lib/prop-injectable");
const custom_resources_1 = require("aws-cdk-lib/custom-resources");
const app_1 = require("./app");
const utils_1 = require("./utils");
const asset_deployment_provider_generated_1 = require("../custom-resource-handlers/dist/aws-amplify-alpha/asset-deployment-provider.generated");
/**
 * An Amplify Console branch
 */
let Branch = (() => {
    let _classDecorators = [prop_injectable_1.propertyInjectable];
    let _classDescriptor;
    let _classExtraInitializers = [];
    let _classThis;
    let _classSuper = core_1.Resource;
    let _instanceExtraInitializers = [];
    let _addEnvironment_decorators;
    var Branch = class extends _classSuper {
        static { _classThis = this; }
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(_classSuper[Symbol.metadata] ?? null) : void 0;
            _addEnvironment_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            __esDecorate(this, null, _addEnvironment_decorators, { kind: "method", name: "addEnvironment", static: false, private: false, access: { has: obj => "addEnvironment" in obj, get: obj => obj.addEnvironment }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(null, _classDescriptor = { value: _classThis }, _classDecorators, { kind: "class", name: _classThis.name, metadata: _metadata }, null, _classExtraInitializers);
            Branch = _classThis = _classDescriptor.value;
            if (_metadata) Object.defineProperty(_classThis, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        static [JSII_RTTI_SYMBOL_1] = { fqn: "@aws-cdk/aws-amplify-alpha.Branch", version: "2.248.0-alpha.0" };
        /** Uniquely identifies this class. */
        static PROPERTY_INJECTION_ID = '@aws-cdk.aws-amplify-alpha.Branch';
        /**
         * Import an existing branch
         */
        static fromBranchName(scope, id, branchName) {
            class Import extends core_1.Resource {
                branchName = branchName;
            }
            return new Import(scope, id);
        }
        /**
         * The ARN of the branch
         *
         * @attribute
         */
        arn = __runInitializers(this, _instanceExtraInitializers);
        branchName;
        environmentVariables;
        constructor(scope, id, props) {
            super(scope, id);
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_BranchProps(props);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, Branch);
                }
                throw error;
            }
            // Enhanced CDK Analytics Telemetry
            (0, metadata_resource_1.addConstructMetadata)(this, props);
            if (props.app instanceof app_1.App) {
                const platform = props.app.platform;
                const isSSR = (0, utils_1.isServerSideRendered)(platform);
                if (props.computeRole && !isSSR) {
                    throw new core_1.ValidationError((0, helpers_internal_1.lit) `InvalidBranchComputeRolePlatform`, '`computeRole` can only be specified for branches of apps with `Platform.WEB_COMPUTE` or `Platform.WEB_DYNAMIC`.', this);
                }
            }
            this.environmentVariables = props.environmentVariables || {};
            const branchName = props.branchName || id;
            const branch = new aws_amplify_1.CfnBranch(this, 'Resource', {
                appId: props.app.appId,
                basicAuthConfig: props.basicAuth && props.basicAuth.bind(this, `${branchName}BasicAuth`),
                branchName,
                buildSpec: props.buildSpec && props.buildSpec.toBuildSpec(),
                description: props.description,
                enableAutoBuild: props.autoBuild ?? true,
                enablePullRequestPreview: props.pullRequestPreview ?? true,
                environmentVariables: core_1.Lazy.any({ produce: () => (0, utils_1.renderEnvironmentVariables)(this.environmentVariables) }, { omitEmptyArray: true }),
                pullRequestEnvironmentName: props.pullRequestEnvironmentName,
                stage: props.stage,
                enablePerformanceMode: props.performanceMode,
                enableSkewProtection: props.skewProtection,
                computeRoleArn: props.computeRole?.roleArn,
            });
            this.arn = branch.attrArn;
            this.branchName = branch.attrBranchName;
            if (props.asset) {
                new core_1.CustomResource(this, 'DeploymentResource', {
                    serviceToken: AmplifyAssetDeploymentProvider.getOrCreate(this),
                    resourceType: 'Custom::AmplifyAssetDeployment',
                    properties: {
                        AppId: props.app.appId,
                        BranchName: branchName,
                        S3ObjectKey: props.asset.s3ObjectKey,
                        S3BucketName: props.asset.s3BucketName,
                    },
                });
            }
        }
        /**
         * Adds an environment variable to this branch.
         *
         * All environment variables that you add are encrypted to prevent rogue
         * access so you can use them to store secret information.
         */
        addEnvironment(name, value) {
            this.environmentVariables[name] = value;
            return this;
        }
        static {
            __runInitializers(_classThis, _classExtraInitializers);
        }
    };
    return Branch = _classThis;
})();
exports.Branch = Branch;
class AmplifyAssetDeploymentProvider extends core_1.NestedStack {
    /**
     * Returns the singleton provider.
     */
    static getOrCreate(scope) {
        const providerId = 'com.amazonaws.cdk.custom-resources.amplify-asset-deployment-provider';
        const stack = core_1.Stack.of(scope);
        const group = stack.node.tryFindChild(providerId) ?? new AmplifyAssetDeploymentProvider(stack, providerId);
        return group.provider.serviceToken;
    }
    provider;
    constructor(scope, id) {
        super(scope, id);
        const onEvent = new asset_deployment_provider_generated_1.AssetDeploymentOnEventFunction(this, 'amplify-asset-deployment-on-event');
        onEvent.addToRolePolicy(new iam.PolicyStatement({
            resources: ['*'],
            actions: [
                's3:GetObject',
                's3:GetSignedUrl',
                'amplify:ListJobs',
                'amplify:StartDeployment',
            ],
        }));
        const isComplete = new asset_deployment_provider_generated_1.AssetDeploymentIsCompleteFunction(this, 'amplify-asset-deployment-is-complete');
        isComplete.addToRolePolicy(new iam.PolicyStatement({
            resources: ['*'],
            actions: ['amplify:GetJob*'],
        }));
        this.provider = new custom_resources_1.Provider(this, 'amplify-asset-deployment-handler-provider', {
            onEventHandler: onEvent,
            isCompleteHandler: isComplete,
            totalTimeout: core_1.Duration.minutes(5),
        });
    }
}
