// @aws-cdk/aws-amplify-alpha/lib/domain.js
"use strict";
var __runInitializers = (this && this.__runInitializers) || function (thisArg, initializers, value) {
    var useValue = arguments.length > 2;
    for (var i = 0; i < initializers.length; i++) {
        value = useValue ? initializers[i].call(thisArg, value) : initializers[i].call(thisArg);
    }
    return useValue ? value : void 0;
};
var __esDecorate = (this && this.__esDecorate) || function (ctor, descriptorIn, decorators, contextIn, initializers, extraInitializers) {
    function accept(f) { if (f !== void 0 && typeof f !== "function") throw new TypeError("Function expected"); return f; }
    var kind = contextIn.kind, key = kind === "getter" ? "get" : kind === "setter" ? "set" : "value";
    var target = !descriptorIn && ctor ? contextIn["static"] ? ctor : ctor.prototype : null;
    var descriptor = descriptorIn || (target ? Object.getOwnPropertyDescriptor(target, contextIn.name) : {});
    var _, done = false;
    for (var i = decorators.length - 1; i >= 0; i--) {
        var context = {};
        for (var p in contextIn) context[p] = p === "access" ? {} : contextIn[p];
        for (var p in contextIn.access) context.access[p] = contextIn.access[p];
        context.addInitializer = function (f) { if (done) throw new TypeError("Cannot add initializers after decoration has completed"); extraInitializers.push(accept(f || null)); };
        var result = (0, decorators[i])(kind === "accessor" ? { get: descriptor.get, set: descriptor.set } : descriptor[key], context);
        if (kind === "accessor") {
            if (result === void 0) continue;
            if (result === null || typeof result !== "object") throw new TypeError("Object expected");
            if (_ = accept(result.get)) descriptor.get = _;
            if (_ = accept(result.set)) descriptor.set = _;
            if (_ = accept(result.init)) initializers.unshift(_);
        }
        else if (_ = accept(result)) {
            if (kind === "field") initializers.unshift(_);
            else descriptor[key] = _;
        }
    }
    if (target) Object.defineProperty(target, contextIn.name, descriptor);
    done = true;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Domain = void 0;
const jsiiDeprecationWarnings = require("../.warnings.jsii.js");
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const aws_amplify_1 = require("aws-cdk-lib/aws-amplify");
const core_1 = require("aws-cdk-lib/core");
const helpers_internal_1 = require("aws-cdk-lib/core/lib/helpers-internal");
const metadata_resource_1 = require("aws-cdk-lib/core/lib/metadata-resource");
const prop_injectable_1 = require("aws-cdk-lib/core/lib/prop-injectable");
/**
 * An Amplify Console domain
 */
let Domain = (() => {
    let _classDecorators = [prop_injectable_1.propertyInjectable];
    let _classDescriptor;
    let _classExtraInitializers = [];
    let _classThis;
    let _classSuper = core_1.Resource;
    let _instanceExtraInitializers = [];
    let _mapSubDomain_decorators;
    let _mapRoot_decorators;
    var Domain = class extends _classSuper {
        static { _classThis = this; }
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(_classSuper[Symbol.metadata] ?? null) : void 0;
            _mapSubDomain_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            _mapRoot_decorators = [(0, metadata_resource_1.MethodMetadata)()];
            __esDecorate(this, null, _mapSubDomain_decorators, { kind: "method", name: "mapSubDomain", static: false, private: false, access: { has: obj => "mapSubDomain" in obj, get: obj => obj.mapSubDomain }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _mapRoot_decorators, { kind: "method", name: "mapRoot", static: false, private: false, access: { has: obj => "mapRoot" in obj, get: obj => obj.mapRoot }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(null, _classDescriptor = { value: _classThis }, _classDecorators, { kind: "class", name: _classThis.name, metadata: _metadata }, null, _classExtraInitializers);
            Domain = _classThis = _classDescriptor.value;
            if (_metadata) Object.defineProperty(_classThis, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        static [JSII_RTTI_SYMBOL_1] = { fqn: "@aws-cdk/aws-amplify-alpha.Domain", version: "2.248.0-alpha.0" };
        /** Uniquely identifies this class. */
        static PROPERTY_INJECTION_ID = '@aws-cdk.aws-amplify-alpha.Domain';
        /**
         * The ARN of the domain
         *
         * @attribute
         */
        arn = __runInitializers(this, _instanceExtraInitializers);
        /**
         * The DNS Record for certificate verification
         *
         * @attribute
         */
        certificateRecord;
        /**
         * The name of the domain
         *
         * @attribute
         */
        domainName;
        /**
         * The status of the domain association
         *
         * @attribute
         */
        domainStatus;
        /**
         * The reason for the current status of the domain
         *
         * @attribute
         */
        statusReason;
        /**
         * Branch patterns for the automatically created subdomain.
         *
         * @attribute
         */
        domainAutoSubDomainCreationPatterns;
        /**
         * The IAM service role for the subdomain.
         *
         * @attribute
         */
        domainAutoSubDomainIamRole;
        /**
         * Specifies whether the automated creation of subdomains for branches is enabled.
         *
         * @attribute
         */
        domainEnableAutoSubDomain;
        subDomains;
        constructor(scope, id, props) {
            super(scope, id);
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_DomainProps(props);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, Domain);
                }
                throw error;
            }
            // Enhanced CDK Analytics Telemetry
            (0, metadata_resource_1.addConstructMetadata)(this, props);
            this.subDomains = props.subDomains || [];
            const domainName = props.domainName || id;
            if (!core_1.Token.isUnresolved(domainName) && domainName.length > 255) {
                throw new core_1.ValidationError((0, helpers_internal_1.lit) `DomainNameTooLong`, `Domain name must be 255 characters or less, got: ${domainName.length} characters.`, this);
            }
            if (!core_1.Token.isUnresolved(domainName) && !/^(((?!-)[A-Za-z0-9-]{0,62}[A-Za-z0-9])\.)+((?!-)[A-Za-z0-9-]{1,62}[A-Za-z0-9])(\.)?$/.test(domainName)) {
                throw new core_1.ValidationError((0, helpers_internal_1.lit) `InvalidDomainName`, `Domain name must be a valid hostname, got: ${domainName}.`, this);
            }
            const domain = new aws_amplify_1.CfnDomain(this, 'Resource', {
                appId: props.app.appId,
                domainName,
                subDomainSettings: core_1.Lazy.any({ produce: () => this.renderSubDomainSettings() }, { omitEmptyArray: true }),
                enableAutoSubDomain: !!props.enableAutoSubdomain,
                autoSubDomainCreationPatterns: props.autoSubdomainCreationPatterns || ['*', 'pr*'],
                autoSubDomainIamRole: props.autoSubDomainIamRole?.roleArn,
                certificateSettings: props.customCertificate ? {
                    certificateType: 'CUSTOM',
                    customCertificateArn: props.customCertificate.certificateArn,
                } : undefined,
            });
            this.arn = domain.attrArn;
            this.certificateRecord = domain.attrCertificateRecord;
            this.domainName = domain.attrDomainName;
            this.domainStatus = domain.attrDomainStatus;
            this.statusReason = domain.attrStatusReason;
            this.domainAutoSubDomainCreationPatterns = domain.attrAutoSubDomainCreationPatterns;
            this.domainAutoSubDomainIamRole = domain.attrAutoSubDomainIamRole;
            this.domainEnableAutoSubDomain = domain.attrEnableAutoSubDomain;
            this.node.addValidation({ validate: () => this.validateDomain() });
        }
        /**
         * Maps a branch to a sub domain
         *
         * @param branch The branch
         * @param prefix The prefix. Use '' to map to the root of the domain. Defaults to branch name.
         */
        mapSubDomain(branch, prefix) {
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_IBranch(branch);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, this.mapSubDomain);
                }
                throw error;
            }
            this.subDomains.push({ branch, prefix });
            return this;
        }
        /**
         * Maps a branch to the domain root
         */
        mapRoot(branch) {
            try {
                jsiiDeprecationWarnings._aws_cdk_aws_amplify_alpha_IBranch(branch);
            }
            catch (error) {
                if (process.env.JSII_DEBUG !== "1" && error.name === "DeprecationError") {
                    Error.captureStackTrace(error, this.mapRoot);
                }
                throw error;
            }
            return this.mapSubDomain(branch, '');
        }
        validateDomain() {
            if (this.subDomains.length === 0) {
                return ['The domain doesn\'t contain any subdomains'];
            }
            return [];
        }
        renderSubDomainSettings() {
            return this.subDomains.map(s => ({
                branchName: s.branch.branchName,
                prefix: s.prefix ?? s.branch.branchName,
            }));
        }
        static {
            __runInitializers(_classThis, _classExtraInitializers);
        }
    };
    return Domain = _classThis;
})();
exports.Domain = Domain;
