// @aws-sdk/client-cloudfront/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooManyCacheBehaviors = exports.RealtimeLogConfigOwnerMismatch = exports.NoSuchResponseHeadersPolicy = exports.NoSuchRealtimeLogConfig = exports.NoSuchOriginRequestPolicy = exports.NoSuchOrigin = exports.NoSuchFieldLevelEncryptionConfig = exports.NoSuchCachePolicy = exports.MissingBody = exports.InvalidWebACLId = exports.InvalidViewerCertificate = exports.InvalidTTLOrder = exports.InvalidResponseCode = exports.InvalidRequiredProtocol = exports.InvalidRelativePath = exports.InvalidQueryStringParameters = exports.InvalidProtocolSettings = exports.InvalidOriginReadTimeout = exports.InvalidOriginKeepaliveTimeout = exports.InvalidOriginAccessIdentity = exports.InvalidOriginAccessControl = exports.InvalidOrigin = exports.InvalidMinimumProtocolVersion = exports.InvalidLocationCode = exports.InvalidLambdaFunctionAssociation = exports.InvalidHeadersForS3Origin = exports.InvalidGeoRestrictionParameter = exports.InvalidFunctionAssociation = exports.InvalidForwardCookies = exports.InvalidErrorCode = exports.InvalidDefaultRootObject = exports.InconsistentQuantities = exports.IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior = exports.DistributionAlreadyExists = exports.CNAMEAlreadyExists = exports.CannotUpdateEntityWhileInUse = exports.CannotDeleteEntityWhileInUse = exports.CannotChangeImmutablePublicKeyFields = exports.CachePolicyInUse = exports.CachePolicyAlreadyExists = exports.BatchTooLarge = exports.PreconditionFailed = exports.InvalidIfMatchVersion = exports.EntityNotFound = exports.EntityLimitExceeded = exports.TooManyDistributionCNAMEs = exports.NoSuchDistribution = exports.InvalidArgument = exports.IllegalUpdate = exports.AccessDenied = void 0;
exports.NoSuchPublicKey = exports.FieldLevelEncryptionProfileSizeExceeded = exports.FieldLevelEncryptionProfileAlreadyExists = exports.TooManyFieldLevelEncryptionQueryArgProfiles = exports.TooManyFieldLevelEncryptionContentTypeProfiles = exports.TooManyFieldLevelEncryptionConfigs = exports.QueryArgProfileEmpty = exports.NoSuchFieldLevelEncryptionProfile = exports.FieldLevelEncryptionConfigAlreadyExists = exports.InvalidAssociation = exports.NoSuchContinuousDeploymentPolicy = exports.InvalidDomainNameForOriginAccessControl = exports.IllegalOriginAccessConfiguration = exports.ContinuousDeploymentPolicyInUse = exports.TooManyContinuousDeploymentPolicies = exports.StagingDistributionInUse = exports.ContinuousDeploymentPolicyAlreadyExists = exports.EntitySizeLimitExceeded = exports.TooManyCloudFrontOriginAccessIdentities = exports.CloudFrontOriginAccessIdentityAlreadyExists = exports.TooManyQueryStringsInCachePolicy = exports.TooManyHeadersInCachePolicy = exports.TooManyCookiesInCachePolicy = exports.TooManyCachePolicies = exports.UnsupportedOperation = exports.InvalidTagging = exports.EntityAlreadyExists = exports.TrustedSignerDoesNotExist = exports.TrustedKeyGroupDoesNotExist = exports.TooManyTrustedSigners = exports.TooManyQueryStringParameters = exports.TooManyOrigins = exports.TooManyOriginGroupsPerDistribution = exports.TooManyOriginCustomHeaders = exports.TooManyLambdaFunctionAssociations = exports.TooManyKeyGroupsAssociatedToDistribution = exports.TooManyHeadersInForwardedValues = exports.TooManyFunctionAssociations = exports.TooManyDistributionsWithSingleFunctionARN = exports.TooManyDistributionsWithLambdaAssociations = exports.TooManyDistributionsWithFunctionAssociations = exports.TooManyDistributionsAssociatedToResponseHeadersPolicy = exports.TooManyDistributionsAssociatedToOriginRequestPolicy = exports.TooManyDistributionsAssociatedToOriginAccessControl = exports.TooManyDistributionsAssociatedToKeyGroup = exports.TooManyDistributionsAssociatedToFieldLevelEncryptionConfig = exports.TooManyDistributionsAssociatedToCachePolicy = exports.TooManyDistributions = exports.TooManyCookieNamesInWhiteList = exports.TooManyCertificates = void 0;
exports.StreamingDistributionNotDisabled = exports.NoSuchStreamingDistribution = exports.ResponseHeadersPolicyInUse = exports.RealtimeLogConfigInUse = exports.PublicKeyInUse = exports.OriginRequestPolicyInUse = exports.OriginAccessControlInUse = exports.NoSuchOriginAccessControl = exports.NoSuchMonitoringSubscription = exports.NoSuchResource = exports.NoSuchFunctionExists = exports.FunctionInUse = exports.FieldLevelEncryptionProfileInUse = exports.FieldLevelEncryptionConfigInUse = exports.ResourceInUse = exports.DistributionNotDisabled = exports.ResourceNotDisabled = exports.NoSuchCloudFrontOriginAccessIdentity = exports.CloudFrontOriginAccessIdentityInUse = exports.IllegalDelete = exports.TooManyStreamingDistributions = exports.TooManyStreamingDistributionCNAMEs = exports.StreamingDistributionAlreadyExists = exports.TooManyResponseHeadersPolicies = exports.TooManyRemoveHeadersInResponseHeadersPolicy = exports.TooManyCustomHeadersInResponseHeadersPolicy = exports.TooLongCSPInResponseHeadersPolicy = exports.ResponseHeadersPolicyAlreadyExists = exports.TooManyRealtimeLogConfigs = exports.RealtimeLogConfigAlreadyExists = exports.TooManyPublicKeys = exports.PublicKeyAlreadyExists = exports.TooManyQueryStringsInOriginRequestPolicy = exports.TooManyOriginRequestPolicies = exports.TooManyHeadersInOriginRequestPolicy = exports.TooManyCookiesInOriginRequestPolicy = exports.OriginRequestPolicyAlreadyExists = exports.TooManyOriginAccessControls = exports.OriginAccessControlAlreadyExists = exports.MonitoringSubscriptionAlreadyExists = exports.TooManyPublicKeysInKeyGroup = exports.TooManyKeyGroups = exports.KeyGroupAlreadyExists = exports.TooManyInvalidationsInProgress = exports.TooManyFunctions = exports.FunctionSizeLimitExceeded = exports.FunctionAlreadyExists = exports.TooManyFieldLevelEncryptionProfiles = exports.TooManyFieldLevelEncryptionFieldPatterns = exports.TooManyFieldLevelEncryptionEncryptionEntities = void 0;
exports.TestFunctionFailed = exports.NoSuchInvalidation = void 0;
const CloudFrontServiceException_1 = require("./CloudFrontServiceException");
class AccessDenied extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "AccessDenied";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AccessDenied",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDenied.prototype);
        this.Message = opts.Message;
    }
}
exports.AccessDenied = AccessDenied;
class IllegalUpdate extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "IllegalUpdate";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalUpdate",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalUpdate.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalUpdate = IllegalUpdate;
class InvalidArgument extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidArgument";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidArgument.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidArgument = InvalidArgument;
class NoSuchDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchDistribution";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchDistribution",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchDistribution = NoSuchDistribution;
class TooManyDistributionCNAMEs extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionCNAMEs";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionCNAMEs",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionCNAMEs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionCNAMEs = TooManyDistributionCNAMEs;
class EntityLimitExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "EntityLimitExceeded";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "EntityLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EntityLimitExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.EntityLimitExceeded = EntityLimitExceeded;
class EntityNotFound extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "EntityNotFound";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "EntityNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EntityNotFound.prototype);
        this.Message = opts.Message;
    }
}
exports.EntityNotFound = EntityNotFound;
class InvalidIfMatchVersion extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidIfMatchVersion";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidIfMatchVersion",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIfMatchVersion.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidIfMatchVersion = InvalidIfMatchVersion;
class PreconditionFailed extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "PreconditionFailed";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PreconditionFailed",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PreconditionFailed.prototype);
        this.Message = opts.Message;
    }
}
exports.PreconditionFailed = PreconditionFailed;
class BatchTooLarge extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "BatchTooLarge";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BatchTooLarge",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchTooLarge.prototype);
        this.Message = opts.Message;
    }
}
exports.BatchTooLarge = BatchTooLarge;
class CachePolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CachePolicyAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CachePolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CachePolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CachePolicyAlreadyExists = CachePolicyAlreadyExists;
class CachePolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CachePolicyInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CachePolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CachePolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CachePolicyInUse = CachePolicyInUse;
class CannotChangeImmutablePublicKeyFields extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CannotChangeImmutablePublicKeyFields";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CannotChangeImmutablePublicKeyFields",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CannotChangeImmutablePublicKeyFields.prototype);
        this.Message = opts.Message;
    }
}
exports.CannotChangeImmutablePublicKeyFields = CannotChangeImmutablePublicKeyFields;
class CannotDeleteEntityWhileInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CannotDeleteEntityWhileInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CannotDeleteEntityWhileInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CannotDeleteEntityWhileInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CannotDeleteEntityWhileInUse = CannotDeleteEntityWhileInUse;
class CannotUpdateEntityWhileInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CannotUpdateEntityWhileInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CannotUpdateEntityWhileInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CannotUpdateEntityWhileInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CannotUpdateEntityWhileInUse = CannotUpdateEntityWhileInUse;
class CNAMEAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CNAMEAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CNAMEAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CNAMEAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CNAMEAlreadyExists = CNAMEAlreadyExists;
class DistributionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "DistributionAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DistributionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DistributionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.DistributionAlreadyExists = DistributionAlreadyExists;
class IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior = IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior;
class InconsistentQuantities extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InconsistentQuantities";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InconsistentQuantities",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InconsistentQuantities.prototype);
        this.Message = opts.Message;
    }
}
exports.InconsistentQuantities = InconsistentQuantities;
class InvalidDefaultRootObject extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidDefaultRootObject";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidDefaultRootObject",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDefaultRootObject.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidDefaultRootObject = InvalidDefaultRootObject;
class InvalidErrorCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidErrorCode";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidErrorCode",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidErrorCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidErrorCode = InvalidErrorCode;
class InvalidForwardCookies extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidForwardCookies";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidForwardCookies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidForwardCookies.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidForwardCookies = InvalidForwardCookies;
class InvalidFunctionAssociation extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidFunctionAssociation";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidFunctionAssociation = InvalidFunctionAssociation;
class InvalidGeoRestrictionParameter extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidGeoRestrictionParameter";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidGeoRestrictionParameter",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidGeoRestrictionParameter.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidGeoRestrictionParameter = InvalidGeoRestrictionParameter;
class InvalidHeadersForS3Origin extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidHeadersForS3Origin";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidHeadersForS3Origin",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidHeadersForS3Origin.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidHeadersForS3Origin = InvalidHeadersForS3Origin;
class InvalidLambdaFunctionAssociation extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidLambdaFunctionAssociation";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidLambdaFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLambdaFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidLambdaFunctionAssociation = InvalidLambdaFunctionAssociation;
class InvalidLocationCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidLocationCode";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidLocationCode",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLocationCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidLocationCode = InvalidLocationCode;
class InvalidMinimumProtocolVersion extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidMinimumProtocolVersion";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidMinimumProtocolVersion",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidMinimumProtocolVersion.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidMinimumProtocolVersion = InvalidMinimumProtocolVersion;
class InvalidOrigin extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidOrigin";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOrigin",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOrigin.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOrigin = InvalidOrigin;
class InvalidOriginAccessControl extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidOriginAccessControl";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginAccessControl = InvalidOriginAccessControl;
class InvalidOriginAccessIdentity extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidOriginAccessIdentity";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOriginAccessIdentity",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOriginAccessIdentity.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginAccessIdentity = InvalidOriginAccessIdentity;
class InvalidOriginKeepaliveTimeout extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidOriginKeepaliveTimeout";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOriginKeepaliveTimeout",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOriginKeepaliveTimeout.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginKeepaliveTimeout = InvalidOriginKeepaliveTimeout;
class InvalidOriginReadTimeout extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidOriginReadTimeout";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOriginReadTimeout",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOriginReadTimeout.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginReadTimeout = InvalidOriginReadTimeout;
class InvalidProtocolSettings extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidProtocolSettings";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidProtocolSettings",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidProtocolSettings.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidProtocolSettings = InvalidProtocolSettings;
class InvalidQueryStringParameters extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidQueryStringParameters";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidQueryStringParameters = InvalidQueryStringParameters;
class InvalidRelativePath extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidRelativePath";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidRelativePath",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRelativePath.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRelativePath = InvalidRelativePath;
class InvalidRequiredProtocol extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidRequiredProtocol";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidRequiredProtocol",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequiredProtocol.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRequiredProtocol = InvalidRequiredProtocol;
class InvalidResponseCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidResponseCode";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidResponseCode",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidResponseCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidResponseCode = InvalidResponseCode;
class InvalidTTLOrder extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidTTLOrder";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidTTLOrder",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTTLOrder.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidTTLOrder = InvalidTTLOrder;
class InvalidViewerCertificate extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidViewerCertificate";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidViewerCertificate",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidViewerCertificate.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidViewerCertificate = InvalidViewerCertificate;
class InvalidWebACLId extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidWebACLId";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidWebACLId",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidWebACLId.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidWebACLId = InvalidWebACLId;
class MissingBody extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "MissingBody";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "MissingBody",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MissingBody.prototype);
        this.Message = opts.Message;
    }
}
exports.MissingBody = MissingBody;
class NoSuchCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchCachePolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCachePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCachePolicy = NoSuchCachePolicy;
class NoSuchFieldLevelEncryptionConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchFieldLevelEncryptionConfig";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFieldLevelEncryptionConfig = NoSuchFieldLevelEncryptionConfig;
class NoSuchOrigin extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchOrigin";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchOrigin",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchOrigin.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchOrigin = NoSuchOrigin;
class NoSuchOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchOriginRequestPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchOriginRequestPolicy = NoSuchOriginRequestPolicy;
class NoSuchRealtimeLogConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchRealtimeLogConfig";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchRealtimeLogConfig",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchRealtimeLogConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchRealtimeLogConfig = NoSuchRealtimeLogConfig;
class NoSuchResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchResponseHeadersPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchResponseHeadersPolicy = NoSuchResponseHeadersPolicy;
class RealtimeLogConfigOwnerMismatch extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "RealtimeLogConfigOwnerMismatch";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RealtimeLogConfigOwnerMismatch",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RealtimeLogConfigOwnerMismatch.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigOwnerMismatch = RealtimeLogConfigOwnerMismatch;
class TooManyCacheBehaviors extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCacheBehaviors";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCacheBehaviors",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCacheBehaviors.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCacheBehaviors = TooManyCacheBehaviors;
class TooManyCertificates extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCertificates";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCertificates",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCertificates.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCertificates = TooManyCertificates;
class TooManyCookieNamesInWhiteList extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCookieNamesInWhiteList";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCookieNamesInWhiteList",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCookieNamesInWhiteList.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookieNamesInWhiteList = TooManyCookieNamesInWhiteList;
class TooManyDistributions extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributions";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributions",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributions = TooManyDistributions;
class TooManyDistributionsAssociatedToCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToCachePolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToCachePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToCachePolicy = TooManyDistributionsAssociatedToCachePolicy;
class TooManyDistributionsAssociatedToFieldLevelEncryptionConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToFieldLevelEncryptionConfig = TooManyDistributionsAssociatedToFieldLevelEncryptionConfig;
class TooManyDistributionsAssociatedToKeyGroup extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToKeyGroup";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToKeyGroup",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToKeyGroup = TooManyDistributionsAssociatedToKeyGroup;
class TooManyDistributionsAssociatedToOriginAccessControl extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToOriginAccessControl";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToOriginAccessControl = TooManyDistributionsAssociatedToOriginAccessControl;
class TooManyDistributionsAssociatedToOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToOriginRequestPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToOriginRequestPolicy = TooManyDistributionsAssociatedToOriginRequestPolicy;
class TooManyDistributionsAssociatedToResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsAssociatedToResponseHeadersPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToResponseHeadersPolicy = TooManyDistributionsAssociatedToResponseHeadersPolicy;
class TooManyDistributionsWithFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsWithFunctionAssociations";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsWithFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithFunctionAssociations = TooManyDistributionsWithFunctionAssociations;
class TooManyDistributionsWithLambdaAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsWithLambdaAssociations";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithLambdaAssociations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsWithLambdaAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithLambdaAssociations = TooManyDistributionsWithLambdaAssociations;
class TooManyDistributionsWithSingleFunctionARN extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyDistributionsWithSingleFunctionARN";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithSingleFunctionARN",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyDistributionsWithSingleFunctionARN.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithSingleFunctionARN = TooManyDistributionsWithSingleFunctionARN;
class TooManyFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFunctionAssociations";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFunctionAssociations = TooManyFunctionAssociations;
class TooManyHeadersInForwardedValues extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyHeadersInForwardedValues";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyHeadersInForwardedValues",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHeadersInForwardedValues.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInForwardedValues = TooManyHeadersInForwardedValues;
class TooManyKeyGroupsAssociatedToDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyKeyGroupsAssociatedToDistribution";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyKeyGroupsAssociatedToDistribution",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyKeyGroupsAssociatedToDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyKeyGroupsAssociatedToDistribution = TooManyKeyGroupsAssociatedToDistribution;
class TooManyLambdaFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyLambdaFunctionAssociations";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyLambdaFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyLambdaFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyLambdaFunctionAssociations = TooManyLambdaFunctionAssociations;
class TooManyOriginCustomHeaders extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyOriginCustomHeaders";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyOriginCustomHeaders",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyOriginCustomHeaders.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginCustomHeaders = TooManyOriginCustomHeaders;
class TooManyOriginGroupsPerDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyOriginGroupsPerDistribution";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyOriginGroupsPerDistribution",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyOriginGroupsPerDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginGroupsPerDistribution = TooManyOriginGroupsPerDistribution;
class TooManyOrigins extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyOrigins";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyOrigins",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyOrigins.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOrigins = TooManyOrigins;
class TooManyQueryStringParameters extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyQueryStringParameters";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringParameters = TooManyQueryStringParameters;
class TooManyTrustedSigners extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyTrustedSigners";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTrustedSigners",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrustedSigners.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyTrustedSigners = TooManyTrustedSigners;
class TrustedKeyGroupDoesNotExist extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TrustedKeyGroupDoesNotExist";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustedKeyGroupDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustedKeyGroupDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
exports.TrustedKeyGroupDoesNotExist = TrustedKeyGroupDoesNotExist;
class TrustedSignerDoesNotExist extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TrustedSignerDoesNotExist";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustedSignerDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustedSignerDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
exports.TrustedSignerDoesNotExist = TrustedSignerDoesNotExist;
class EntityAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "EntityAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "EntityAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EntityAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.EntityAlreadyExists = EntityAlreadyExists;
class InvalidTagging extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidTagging";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidTagging",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTagging.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidTagging = InvalidTagging;
class UnsupportedOperation extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "UnsupportedOperation";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "UnsupportedOperation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedOperation.prototype);
        this.Message = opts.Message;
    }
}
exports.UnsupportedOperation = UnsupportedOperation;
class TooManyCachePolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCachePolicies";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCachePolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCachePolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCachePolicies = TooManyCachePolicies;
class TooManyCookiesInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCookiesInCachePolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCookiesInCachePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCookiesInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookiesInCachePolicy = TooManyCookiesInCachePolicy;
class TooManyHeadersInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyHeadersInCachePolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyHeadersInCachePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHeadersInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInCachePolicy = TooManyHeadersInCachePolicy;
class TooManyQueryStringsInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyQueryStringsInCachePolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInCachePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyQueryStringsInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringsInCachePolicy = TooManyQueryStringsInCachePolicy;
class CloudFrontOriginAccessIdentityAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CloudFrontOriginAccessIdentityAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CloudFrontOriginAccessIdentityAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CloudFrontOriginAccessIdentityAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CloudFrontOriginAccessIdentityAlreadyExists = CloudFrontOriginAccessIdentityAlreadyExists;
class TooManyCloudFrontOriginAccessIdentities extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCloudFrontOriginAccessIdentities";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCloudFrontOriginAccessIdentities",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCloudFrontOriginAccessIdentities.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCloudFrontOriginAccessIdentities = TooManyCloudFrontOriginAccessIdentities;
class EntitySizeLimitExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "EntitySizeLimitExceeded";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "EntitySizeLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EntitySizeLimitExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.EntitySizeLimitExceeded = EntitySizeLimitExceeded;
class ContinuousDeploymentPolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ContinuousDeploymentPolicyAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ContinuousDeploymentPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ContinuousDeploymentPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.ContinuousDeploymentPolicyAlreadyExists = ContinuousDeploymentPolicyAlreadyExists;
class StagingDistributionInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "StagingDistributionInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StagingDistributionInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StagingDistributionInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.StagingDistributionInUse = StagingDistributionInUse;
class TooManyContinuousDeploymentPolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyContinuousDeploymentPolicies";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyContinuousDeploymentPolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyContinuousDeploymentPolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyContinuousDeploymentPolicies = TooManyContinuousDeploymentPolicies;
class ContinuousDeploymentPolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ContinuousDeploymentPolicyInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ContinuousDeploymentPolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ContinuousDeploymentPolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.ContinuousDeploymentPolicyInUse = ContinuousDeploymentPolicyInUse;
class IllegalOriginAccessConfiguration extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "IllegalOriginAccessConfiguration";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalOriginAccessConfiguration",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalOriginAccessConfiguration.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalOriginAccessConfiguration = IllegalOriginAccessConfiguration;
class InvalidDomainNameForOriginAccessControl extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidDomainNameForOriginAccessControl";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidDomainNameForOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDomainNameForOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidDomainNameForOriginAccessControl = InvalidDomainNameForOriginAccessControl;
class NoSuchContinuousDeploymentPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchContinuousDeploymentPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchContinuousDeploymentPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchContinuousDeploymentPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchContinuousDeploymentPolicy = NoSuchContinuousDeploymentPolicy;
class InvalidAssociation extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "InvalidAssociation";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAssociation.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidAssociation = InvalidAssociation;
class FieldLevelEncryptionConfigAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FieldLevelEncryptionConfigAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FieldLevelEncryptionConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionConfigAlreadyExists = FieldLevelEncryptionConfigAlreadyExists;
class NoSuchFieldLevelEncryptionProfile extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchFieldLevelEncryptionProfile";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionProfile",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionProfile.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFieldLevelEncryptionProfile = NoSuchFieldLevelEncryptionProfile;
class QueryArgProfileEmpty extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "QueryArgProfileEmpty";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "QueryArgProfileEmpty",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueryArgProfileEmpty.prototype);
        this.Message = opts.Message;
    }
}
exports.QueryArgProfileEmpty = QueryArgProfileEmpty;
class TooManyFieldLevelEncryptionConfigs extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionConfigs";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionConfigs",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionConfigs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionConfigs = TooManyFieldLevelEncryptionConfigs;
class TooManyFieldLevelEncryptionContentTypeProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionContentTypeProfiles";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionContentTypeProfiles",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionContentTypeProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionContentTypeProfiles = TooManyFieldLevelEncryptionContentTypeProfiles;
class TooManyFieldLevelEncryptionQueryArgProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionQueryArgProfiles";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionQueryArgProfiles",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionQueryArgProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionQueryArgProfiles = TooManyFieldLevelEncryptionQueryArgProfiles;
class FieldLevelEncryptionProfileAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FieldLevelEncryptionProfileAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileAlreadyExists = FieldLevelEncryptionProfileAlreadyExists;
class FieldLevelEncryptionProfileSizeExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FieldLevelEncryptionProfileSizeExceeded";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileSizeExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileSizeExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileSizeExceeded = FieldLevelEncryptionProfileSizeExceeded;
class NoSuchPublicKey extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchPublicKey";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchPublicKey",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchPublicKey.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchPublicKey = NoSuchPublicKey;
class TooManyFieldLevelEncryptionEncryptionEntities extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionEncryptionEntities";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionEncryptionEntities",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionEncryptionEntities.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionEncryptionEntities = TooManyFieldLevelEncryptionEncryptionEntities;
class TooManyFieldLevelEncryptionFieldPatterns extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionFieldPatterns";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionFieldPatterns",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionFieldPatterns.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionFieldPatterns = TooManyFieldLevelEncryptionFieldPatterns;
class TooManyFieldLevelEncryptionProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFieldLevelEncryptionProfiles";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionProfiles",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionProfiles = TooManyFieldLevelEncryptionProfiles;
class FunctionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FunctionAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FunctionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FunctionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionAlreadyExists = FunctionAlreadyExists;
class FunctionSizeLimitExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FunctionSizeLimitExceeded";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FunctionSizeLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FunctionSizeLimitExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionSizeLimitExceeded = FunctionSizeLimitExceeded;
class TooManyFunctions extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyFunctions";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyFunctions",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyFunctions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFunctions = TooManyFunctions;
class TooManyInvalidationsInProgress extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyInvalidationsInProgress";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyInvalidationsInProgress",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyInvalidationsInProgress.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyInvalidationsInProgress = TooManyInvalidationsInProgress;
class KeyGroupAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "KeyGroupAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "KeyGroupAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeyGroupAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.KeyGroupAlreadyExists = KeyGroupAlreadyExists;
class TooManyKeyGroups extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyKeyGroups";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyKeyGroups",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyKeyGroups.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyKeyGroups = TooManyKeyGroups;
class TooManyPublicKeysInKeyGroup extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyPublicKeysInKeyGroup";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyPublicKeysInKeyGroup",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyPublicKeysInKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyPublicKeysInKeyGroup = TooManyPublicKeysInKeyGroup;
class MonitoringSubscriptionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "MonitoringSubscriptionAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "MonitoringSubscriptionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MonitoringSubscriptionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.MonitoringSubscriptionAlreadyExists = MonitoringSubscriptionAlreadyExists;
class OriginAccessControlAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "OriginAccessControlAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OriginAccessControlAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OriginAccessControlAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginAccessControlAlreadyExists = OriginAccessControlAlreadyExists;
class TooManyOriginAccessControls extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyOriginAccessControls";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyOriginAccessControls",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyOriginAccessControls.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginAccessControls = TooManyOriginAccessControls;
class OriginRequestPolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "OriginRequestPolicyAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OriginRequestPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OriginRequestPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginRequestPolicyAlreadyExists = OriginRequestPolicyAlreadyExists;
class TooManyCookiesInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCookiesInOriginRequestPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCookiesInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCookiesInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookiesInOriginRequestPolicy = TooManyCookiesInOriginRequestPolicy;
class TooManyHeadersInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyHeadersInOriginRequestPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyHeadersInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHeadersInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInOriginRequestPolicy = TooManyHeadersInOriginRequestPolicy;
class TooManyOriginRequestPolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyOriginRequestPolicies";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyOriginRequestPolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyOriginRequestPolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginRequestPolicies = TooManyOriginRequestPolicies;
class TooManyQueryStringsInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyQueryStringsInOriginRequestPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyQueryStringsInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringsInOriginRequestPolicy = TooManyQueryStringsInOriginRequestPolicy;
class PublicKeyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "PublicKeyAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PublicKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PublicKeyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.PublicKeyAlreadyExists = PublicKeyAlreadyExists;
class TooManyPublicKeys extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyPublicKeys";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyPublicKeys",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyPublicKeys.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyPublicKeys = TooManyPublicKeys;
class RealtimeLogConfigAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "RealtimeLogConfigAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RealtimeLogConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RealtimeLogConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigAlreadyExists = RealtimeLogConfigAlreadyExists;
class TooManyRealtimeLogConfigs extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyRealtimeLogConfigs";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRealtimeLogConfigs",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRealtimeLogConfigs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRealtimeLogConfigs = TooManyRealtimeLogConfigs;
class ResponseHeadersPolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ResponseHeadersPolicyAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResponseHeadersPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResponseHeadersPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.ResponseHeadersPolicyAlreadyExists = ResponseHeadersPolicyAlreadyExists;
class TooLongCSPInResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooLongCSPInResponseHeadersPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooLongCSPInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooLongCSPInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooLongCSPInResponseHeadersPolicy = TooLongCSPInResponseHeadersPolicy;
class TooManyCustomHeadersInResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyCustomHeadersInResponseHeadersPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCustomHeadersInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCustomHeadersInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCustomHeadersInResponseHeadersPolicy = TooManyCustomHeadersInResponseHeadersPolicy;
class TooManyRemoveHeadersInResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyRemoveHeadersInResponseHeadersPolicy";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRemoveHeadersInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRemoveHeadersInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRemoveHeadersInResponseHeadersPolicy = TooManyRemoveHeadersInResponseHeadersPolicy;
class TooManyResponseHeadersPolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyResponseHeadersPolicies";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyResponseHeadersPolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyResponseHeadersPolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyResponseHeadersPolicies = TooManyResponseHeadersPolicies;
class StreamingDistributionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "StreamingDistributionAlreadyExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StreamingDistributionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StreamingDistributionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.StreamingDistributionAlreadyExists = StreamingDistributionAlreadyExists;
class TooManyStreamingDistributionCNAMEs extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyStreamingDistributionCNAMEs";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyStreamingDistributionCNAMEs",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyStreamingDistributionCNAMEs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyStreamingDistributionCNAMEs = TooManyStreamingDistributionCNAMEs;
class TooManyStreamingDistributions extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TooManyStreamingDistributions";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyStreamingDistributions",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyStreamingDistributions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyStreamingDistributions = TooManyStreamingDistributions;
class IllegalDelete extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "IllegalDelete";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalDelete",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalDelete.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalDelete = IllegalDelete;
class CloudFrontOriginAccessIdentityInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "CloudFrontOriginAccessIdentityInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CloudFrontOriginAccessIdentityInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CloudFrontOriginAccessIdentityInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CloudFrontOriginAccessIdentityInUse = CloudFrontOriginAccessIdentityInUse;
class NoSuchCloudFrontOriginAccessIdentity extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchCloudFrontOriginAccessIdentity";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCloudFrontOriginAccessIdentity",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCloudFrontOriginAccessIdentity.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCloudFrontOriginAccessIdentity = NoSuchCloudFrontOriginAccessIdentity;
class ResourceNotDisabled extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ResourceNotDisabled";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotDisabled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotDisabled.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceNotDisabled = ResourceNotDisabled;
class DistributionNotDisabled extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "DistributionNotDisabled";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DistributionNotDisabled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DistributionNotDisabled.prototype);
        this.Message = opts.Message;
    }
}
exports.DistributionNotDisabled = DistributionNotDisabled;
class ResourceInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ResourceInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceInUse = ResourceInUse;
class FieldLevelEncryptionConfigInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FieldLevelEncryptionConfigInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionConfigInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FieldLevelEncryptionConfigInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionConfigInUse = FieldLevelEncryptionConfigInUse;
class FieldLevelEncryptionProfileInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FieldLevelEncryptionProfileInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileInUse = FieldLevelEncryptionProfileInUse;
class FunctionInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "FunctionInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FunctionInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FunctionInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionInUse = FunctionInUse;
class NoSuchFunctionExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchFunctionExists";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchFunctionExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchFunctionExists.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFunctionExists = NoSuchFunctionExists;
class NoSuchResource extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchResource";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchResource",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchResource.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchResource = NoSuchResource;
class NoSuchMonitoringSubscription extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchMonitoringSubscription";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchMonitoringSubscription",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchMonitoringSubscription.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchMonitoringSubscription = NoSuchMonitoringSubscription;
class NoSuchOriginAccessControl extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchOriginAccessControl";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchOriginAccessControl = NoSuchOriginAccessControl;
class OriginAccessControlInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "OriginAccessControlInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OriginAccessControlInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OriginAccessControlInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginAccessControlInUse = OriginAccessControlInUse;
class OriginRequestPolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "OriginRequestPolicyInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OriginRequestPolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OriginRequestPolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginRequestPolicyInUse = OriginRequestPolicyInUse;
class PublicKeyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "PublicKeyInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PublicKeyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PublicKeyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.PublicKeyInUse = PublicKeyInUse;
class RealtimeLogConfigInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "RealtimeLogConfigInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RealtimeLogConfigInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RealtimeLogConfigInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigInUse = RealtimeLogConfigInUse;
class ResponseHeadersPolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "ResponseHeadersPolicyInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResponseHeadersPolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResponseHeadersPolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.ResponseHeadersPolicyInUse = ResponseHeadersPolicyInUse;
class NoSuchStreamingDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchStreamingDistribution";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchStreamingDistribution",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchStreamingDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchStreamingDistribution = NoSuchStreamingDistribution;
class StreamingDistributionNotDisabled extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "StreamingDistributionNotDisabled";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StreamingDistributionNotDisabled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StreamingDistributionNotDisabled.prototype);
        this.Message = opts.Message;
    }
}
exports.StreamingDistributionNotDisabled = StreamingDistributionNotDisabled;
class NoSuchInvalidation extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "NoSuchInvalidation";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchInvalidation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchInvalidation.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchInvalidation = NoSuchInvalidation;
class TestFunctionFailed extends CloudFrontServiceException_1.CloudFrontServiceException {
    name = "TestFunctionFailed";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "TestFunctionFailed",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, TestFunctionFailed.prototype);
        this.Message = opts.Message;
    }
}
exports.TestFunctionFailed = TestFunctionFailed;
