// aws-cdk/lib/cxapp/exec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.execProgram = execProgram;
exports.createAssembly = createAssembly;
const path = require("path");
const util_1 = require("util");
const cloud_assembly_api_1 = require("@aws-cdk/cloud-assembly-api");
const cxschema = require("@aws-cdk/cloud-assembly-schema");
const cxapi = require("@aws-cdk/cx-api");
const toolkit_lib_1 = require("@aws-cdk/toolkit-lib");
const fs = require("fs-extra");
const api_1 = require("../api");
const user_configuration_1 = require("../cli/user-configuration");
const fs_util_1 = require("../cli/util/fs-util");
const version_1 = require("../cli/version");
/** Invokes the cloud executable and returns JSON output */
async function execProgram(aws, ioHelper, config) {
    const debugFn = (msg) => ioHelper.defaults.debug(msg);
    let errorFile;
    const params = (0, api_1.synthParametersFromSettings)(config.settings);
    const context = {
        ...config.context.all,
        ...params.context,
    };
    await debugFn((0, util_1.format)('context:', context));
    if (params.env.CDK_DEBUG === 'true') {
        await ioHelper.defaults.info('🔍 Synthesizing with debug information. This may take a bit longer.');
    }
    const env = noUndefined({
        // Versioning, outdir, default account and region
        ...await (0, api_1.prepareDefaultEnvironment)(aws, debugFn),
        // Environment variables derived from settings
        ...params.env,
    });
    const build = config.settings.get(['build']);
    if (build) {
        await exec(build);
    }
    let app = config.settings.get(['app']);
    if (!app) {
        throw new toolkit_lib_1.ToolkitError('AppRequired', `--app is required either in command-line, in ${user_configuration_1.PROJECT_CONFIG} or in ${user_configuration_1.USER_DEFAULTS}`);
    }
    // bypass "synth" if app points to a cloud assembly
    if (await fs.pathExists(app) && (await fs.stat(app)).isDirectory()) {
        await debugFn('--app points to a cloud assembly, so we bypass synth');
        // Acquire a read lock on this directory
        const lock = await new api_1.RWLock(app).acquireRead();
        return { assembly: createAssembly(app), lock };
    }
    // Traditionally it has been possible, though not widely advertised, to put a string[] into `cdk.json`.
    // However, we would just quickly join this array back up to string with spaces (unquoted even!) and proceed as usual,
    // thereby losing all the benefits of a pre-segmented command line. This coercion is just here for backwards
    // compatibility with existing configurations. An upcoming PR might retain the benefit of the string[].
    if (Array.isArray(app)) {
        app = app.join(' ');
    }
    const commandLine = await (0, api_1.guessExecutable)(app, debugFn);
    const outdir = config.settings.get(['output']);
    if (!outdir) {
        throw new toolkit_lib_1.ToolkitError('OutputRequired', 'unexpected: --output is required');
    }
    if (typeof outdir !== 'string') {
        throw new toolkit_lib_1.ToolkitError('OutputNotString', `--output takes a string, got ${JSON.stringify(outdir)}`);
    }
    try {
        await fs.mkdirp(outdir);
    }
    catch (error) {
        throw new toolkit_lib_1.ToolkitError('OutputDirCreateFailed', `Could not create output directory ${outdir} (${error.message})`);
    }
    await debugFn(`outdir: ${outdir}`);
    env[cxapi.OUTDIR_ENV] = outdir;
    // Send version information
    env[cxapi.CLI_ASM_VERSION_ENV] = cxschema.Manifest.version();
    env[cxapi.CLI_VERSION_ENV] = (0, version_1.versionNumber)();
    // Acquire a lock on the output directory
    const writerLock = await new api_1.RWLock(outdir).acquireWrite();
    // Prepare an errorFile location
    errorFile = path.join(outdir, 'error.txt');
    await fs.promises.rm(errorFile, { force: true });
    env.CDK_ERROR_FILE = errorFile;
    // Prepare a performance counters location
    const perfCountersFile = path.join(outdir, 'performance-counters.json');
    await fs.promises.rm(perfCountersFile, { force: true });
    env.CDK_PERF_COUNTERS_FILE = perfCountersFile;
    await debugFn((0, util_1.format)('env:', env));
    const cleanupTemp = (0, api_1.writeContextToEnv)(env, context, 'add-process-env-later');
    try {
        await exec(commandLine);
        const assembly = createAssembly(outdir);
        const perfCounters = await (0, fs_util_1.tryReadJson)(perfCountersFile);
        return {
            assembly,
            lock: await writerLock.convertToReaderLock(),
            perfCounters: typeof perfCounters === 'object' && typeof perfCounters.counters === 'object' ? perfCounters.counters : undefined,
        };
    }
    catch (e) {
        await writerLock.release();
        throw e;
    }
    finally {
        await cleanupTemp();
    }
    async function exec(commandAndArgs) {
        try {
            return await (0, api_1.execInChildProcess)(commandAndArgs, {
                env: {
                    ...process.env,
                    ...env,
                },
                errorCodeFile: errorFile,
                // Not capturing the stdout/stderr of the CDK app. It must remain attached to a terminal
                // if the parent process is attached to a terminal.
                captureOutput: false,
            });
        }
        catch (e) {
            await debugFn(`failed command: ${commandAndArgs}`);
            throw e;
        }
    }
}
/**
 * Creates an assembly with error handling
 */
function createAssembly(appDir) {
    try {
        return new cloud_assembly_api_1.CloudAssembly(appDir, {
            // We sort as we deploy
            topoSort: false,
        });
    }
    catch (error) {
        if (error.message.includes(cxschema.VERSION_MISMATCH)) {
            // this means the CLI version is too old.
            // we instruct the user to upgrade.
            throw new toolkit_lib_1.ToolkitError('CliVersionMismatch', `This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version.\n(${error.message})`);
        }
        throw error;
    }
}
function noUndefined(xs) {
    return Object.fromEntries(Object.entries(xs).filter(([_, v]) => v !== undefined));
}
